src/executor/executor.ts

```diff
@@ -117,7 +117,7 @@ export class TaskExecutor {
   private networkService?: NetworkService;
   private statsService: StatsService;
   private activityReadySetupFunctions: Worker[] = [];
-  private taskQueue: TaskQueue<Task<unknown, unknown>>;
+  private taskQueue: TaskQueue<Task>;
```

Comment:
One nice improvement can be to change the signature to `TaskQueue<T extends QueueableTask = Task>` to provide a nice default Then we could just use `private taskQueue: TaskQueue;` instead.